### ISSUE TITLE
Keyboard Trouble. Fn as Meta, but then can't access other characters.

### ISSUE THREAD

**Issue by wretanhewe** [gemian/gemian#20]
I just acquired a used Cosmo Communicator, and have installed and started trying to use Gemian on it. My intent is (since it can make phonecalls and send texts) to try to migrate to using this moreso than android.
However, I just ran into an issue that I was not expecting. 

On the keyboard, the "Fn" button is picked up as a meta/super key, which at first was great! but then I realized that some keys that I often need (underscore, plus sign, page up/down, home, end, commas, quotation marks, and parentheses) are all "function" keys that I can't access now. (But even though it functions as a meta key, somehow it also doesn't always get registered as a meta key, when I try to do specific shortcuts.)

I suppose an alternative could be trying to make some global shortcut for "meta+O" or "meta+I" to send the keystrokes for "_" or "+", but I'm not finding any good ways to do that. I can absolutely manage without a super/meta key. Important global shortcuts I can re-create with (or they already use) ctrl, alt, or shift keys. But I can't find a good way to either 'disable' the super/meta key (in hopes that the Fn key would then work as Planet Computers designed it on the keyboard), nor to make a custom shortcut that will input the "Special character" of a standard underscore or plus sign.

**Comment by adamboardman** [comment]
Just to check that you are using Gemian v4 and at installation time you picked your keyboard layout? Doing this should result in the keyboard working fully eg Fn+J => underscore, Fn+Up => PgUp etc. I have found that the PgUp/PgDn keys work in some apps but not all.

**Comment by dnapla** [comment]
@adamboardman . I've also got this occuring on my version too. It looks the Fn key function has reallocated.... I will try to work out whats going on and let you know. (This is basically a fresh V4 with the latest sudo apt upgrades + new patched ril only)

**Comment by adamboardman** [comment]
That's messed up, so when did it start failing (v4, v4+apt, v4+something else)? Also which keyboard layout did you pick to start with? Did you ever interact with the KDE keyboard settings widget (recommend against - suspect possible cause of problems)?
To help debug where the problems are you can run 'xev' to see keypress events.
Also check that /etc/default/keyboard contains the correct layout.
Also which kernel version, eg output from 'uname -a'.

**Comment by dnapla** [comment]
I picked GB layout. 
I think your second point is correct. I know I opened it to have a look around and see what was available... (I'm don't think I actually changed anything (intentionally) but I toggled some dialogs on/off to see options)

/etc/default/keyboard
XKBLAYOUT=gb
BACKSPACE=guess

Linux cosmocom 4.4.146 #1 SMP PREEMPT Mon Apr 19 21:45:57 UTC 2021 aarch64 GNU/Linux

Do you know a way to revert - or where the widget makes changes?

**Comment by adamboardman** [comment]
Default keyboard layout looks good, suggest adding a new user and then rebooting and logging into that user then if that works again compare the dot files in your two user accounts to see what's changed.

**Comment by adamboardman** [comment]
Just occurred to me that there could be more than one way to add a new user, 'sudo adduser <name>' is the way I do it as that way you get the new user added to all the right permissions groups.

**Comment by dnapla** [comment]
@wretanhewe 

In .config/plasma-localerc - non working 
[Formats]
LANG=**C.UTF-8**

Should be (assuming GB keyboard)
[Formats]
LANG=**en_GB.utf8**

So something about playing in system settings (either shortcuts \ keyboards) changes this

**Comment by wretanhewe** [comment]
Thanks for the replies! (somehow Github was just completely not notifying me of them...)
Since my post, I've wiped and reinstalled Gemian about 5 times, due to other use-breaking problems that I've not been able to track down yet. (I'll probably make another post or two on these if they keep arising. Mainly desktop session issues like cursor moving, but nothing is interactable at all, requiring a hard restart. Or when the screen locks, theres no login prompt which also requires hard restarts after each time I close the device... only way to really 'get in' at the time was a full reinstall while I figure out how all this works)

Anyway, after the reinstalls, the keyboard has been fine, and hasn't gone wrong again (yet).

Checking right now though (while it is working fine), I am running version 4.4.146
.config/plasma-localerc has 
[Formats]
LANG=en_US.utf8
(since Im in the USA)

and /etc/default/keyboard has 
XKBLAYOUT="us"
BACKSPACE="guess"

Back when the issue happened, I'd installed Gemian, selected the US keyboard layout, and I think things were fine at the very beginning. I'd switched to dark theme, tried phonecalls/texts, and installed Telegram and KDE connect from the built in "Discover" software center. I also swapped the application launcher with one that's a fullscreen dashboard launcher, from the KDE widget store that's built-in to KDE. None of these strike me as being quite so big of changes as to mess up the keyboard functionality so enormously, but something might have. 
Also, while the built in keyboard had these issues, I could connect a standard full size  USB keyboard with my Hub, and it connected and functioned all like it should have. 

It's working now though, and if it happens again, I'll take a look at these files and make sure they're working right.

@dnapla , were you able to get yours functioning? This issue might be closeable now.
